src/moonlink/src/storage/mooncake_table/persistence_buffer.rs

```diff
@@ -88,15 +88,25 @@ impl UnpersistedRecords {
 
     /// Get unpersisted data files as hash set for lookup.
     pub(crate) fn get_unpersisted_data_files_set(&self) -> HashSet<MooncakeDataFileRef> {
-        self.new_data_files.iter().cloned().collect::<HashSet<_>>()
+        let expected_len = self.new_data_files.len() + self.compacted_data_files_to_add.len();
```

Comment:
This was a bug: it's used to decide which file indices could be used to perform index merge.
Currently index merge is only attempted upon persisted ones, so need to consider all unpersisted items; I missed compacted results.